nonos/main.py

```diff
@@ -680,503 +737,201 @@ def plot(
+        # We look for the index of the 3d dimension 1D array
+        # that matches self._default_point
+        if average:
+            data = choosemean(self.data, plane, self.coord)
+        else:
+            data = chooseslice(self.data, plane, self.coord, self._default_point)
+
+        # convert 1D coordinates arrays (self.coord)
+        # into 2D coordinates arrays (coordgrid) via meshgrid,
+        # using (plane[0],plane[1]) for the projection plane
+        # and self._default_point for the 3d (plane[2]) dimension
+        coordgrid = meshgrid_from_plane(
+            self.coord, plane[0], plane[1], self._default_point
         )
-
-    def get_v(self, v, x, y):
-        """
-        For a real set of coordinates (x,y), returns the bilinear
-        interpolated value of a Field class.
-        """
-
-        i = find_nearest(self.x, x)
-        # i = int(np.log10(x/self.x.min())/np.log10(self.x.max()/self.x.min())*self.nx)
-        # i = int((x-self.x.min())/(self.x.max()-self.x.min())*self.nx)
-        j = int((y - self.y.min()) / (self.y.max() - self.y.min()) * self.ny)
-
-        if i < 0 or j < 0 or i > v.shape[0] - 2 or j > v.shape[1] - 2:
-            return None
-
-        f11 = v[i, j, self.imidplane]
-        f12 = v[i, j + 1, self.imidplane]
-        f21 = v[i + 1, j, self.imidplane]
-        f22 = v[i + 1, j + 1, self.imidplane]
-        try:
-            x1 = self.x[i]
-            x2 = self.x[i + 1]
-            y1 = self.y[j]
-            y2 = self.y[j + 1]
-            return self.bilinear((x1, x2), (y1, y2), (f11, f12, f21, f22), (x, y))
-        except IndexError:
-            return None
-
-    def euler(self, vx, vy, x, y, reverse):
-        """
-        Euler integrator for computing the streamlines.
-        Parameters:
-        ----------
-
-        x,y: Floats.
-             Initial condition
-        reverse: Boolean.
-                 If reverse is true, the integration step is negative.
-
-        Output
-        ------
-
-        (dx,dy): (float,float).
-                 Are the azimutal and radial increments.
-                 Only works for cylindrical coordinates.
-        """
-        sign = 1.0
-        if reverse:
-            sign = -1
-        vr = self.get_v(vx, x, y)
-        vt = self.get_v(vy, x, y)
-        if None in (vt, vr):
-            # Avoiding problems...
-            return None, None
-
-        l = np.min(
-            (
-                ((self.x.max() - self.x.min()) / self.nx),
-                ((self.y.max() - self.y.min()) / self.ny),
+        coordgrid = np.array(coordgrid, dtype=object)[np.array(plane) - 1]
+
+        # (coord[0]=R,coord[1]=phi by default)
+        # then, depending on the geometry,
+        # we transform these 2D coordinates arrays
+        transform = coordgrid
+        if func_proj is not None:
+            transform = func_proj(
+                coordgrid[0],
+                coordgrid[1],
+                coordgrid[2],
```

Comment:
strictly equivalent (unless if `coordgrid` can ever be longer than 3 rows)
```suggestion
            transform = func_proj(*coordgrid)
```